test-cypress/cypress/commands/api/recording.ts

```diff
@@ -0,0 +1,132 @@
+// load the global Cypress types
+/// <reference types="cypress" />
+
+import {v1ApiPath, uploadFile, DEFAULT_DATE } from "../server";
+import {logTestDescription} from "../descriptions"; 
+
+let lastUsedTime = DEFAULT_DATE;
+
+Cypress.Commands.add("uploadRecording", (cameraName: string, details: thermalRecordingInfo, log = true) => {
+  const data = makeRecordingDataFromDetails(details);
+
+  logTestDescription(`Upload '${JSON.stringify(details)}' recording to '${cameraName}'`,
+    { camera : cameraName, requestData: data},
+    log
+  );
+
+  const fileName = 'invalid.cptv';
+  const url = v1ApiPath('recordings');
+  const fileType = 'application/cptv';
+
+  uploadFile(url, cameraName, fileName, fileType, data);
+  // must wait until the upload request has completed
+  cy.wait(['@addRecording']);
+});
+
+interface trackData {
+  start_s? : number,
+  end_s? : number,
+  confident_tag?: string,
+  confidence?: number,
+}
+
+interface algorithmMetadata {
+  model_name? : string
+}
+
+interface thermalRecordingMetaData {
+  algorithm? : algorithmMetadata,
+  tracks : trackData[]
+}
+
+interface thermalRecordingData {
+  type: "thermalRaw",
+  recordingDateTime: string, 
+  duration: number, 
+  comment? : string,
+  batteryLevel? : number, 
+  batteryCharging? : string, 
```

Comment:
We can do better with types like this: see `Recording.ts:210` - there are a known set of allowed strings for `batteryCharging`.  If we're reusing that set often, we could declare it as an enum.